subworkflows/local/euk_mags_generation.nf

```diff
@@ -127,22 +38,18 @@ workflow EUK_MAGS_GENERATION {
 
     /* split the inputs */
     assemblies_reads_bins.multiMap { meta, assembly, reads, concoct_bins, metabat_bins ->
-        assembly_and_reads: [ meta, assembly, reads ]
+        assembly: [ meta, assembly ]
         reads: [ meta, reads ]
         bins_concoct: [ meta, concoct_bins ]
         bins_metabat: [ meta, metabat_bins ]
     }.set {
         input
     }
```

Comment:
In this block of code you are using the same meta.id in all your tuples? I am wondering if you are losing the `meta.single` in the reads tuple. if that's true, then you are processing single-end as paired-end reads. Not sure if you are using the `meta.single` label in further processes other than decontamination.

Reply:
I'm using whole meta object. It has `.single`, `.erz` and `.id`

Reply:
Brilliant!